### ISSUE TITLE
Type hints are broken

### ISSUE THREAD

**Issue by Take-Some-Bytes** [Take-Some-Bytes/python_http_parser#3]
Running [``mypy``](https://github.com/python/mypy) on the package results in something like this:
```log
python_http_parser/body.py:32: error: Cannot assign to a type
python_http_parser/body.py:32: error: Incompatible types in assignment (expression has type "Type[Mapping[Any, Any]]", variable has type "Type[BodyProcessorCallbacks]")
python_http_parser/body.py:173: error: Incompatible types in assignment (expression has type "int", variable has type "None")
python_http_parser/body.py:203: error: Unsupported operand types for < ("int" and "None")
... More errors ...
Found 38 errors in 2 files (checked 10 source files)
```

Which means, in short, that users won't be able to use a static type checker with the package.

Also, files that use ``.pyi`` annotations don't seem to work with ``mypy``.

**Comment by Take-Some-Bytes** [comment]
Actually, stub files _do_ work with ``mypy``.... That's one problem taken care of.

**Comment by Take-Some-Bytes** [comment]
Fixed by release [v0.4.1](https://github.com/Take-Some-Bytes/python_http_parser/blob/main/CHANGELOG.rst#v041---2021-07-31)
